clients/adapter/src/main/java/org/eclipse/hono/adapter/client/command/InternalCommandSender.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.adapter.client.command;
+
+import org.eclipse.hono.util.Lifecycle;
+
+import io.vertx.core.Future;
+
+/**
+ * A client to send commands to the internal Command and Control API endpoint provided by protocol adapters.
+ */
+public interface InternalCommandSender extends Lifecycle {
+
+    /**
+     * Sends a command to the downstream peer, targeting the protocol adapter Command and Control API endpoint
+     * identified by the given protocol adapter instance id.
+     * <p>
+     * Note that the result of the send operation is getting applied to the command context here.
```

Comment:
It is not, unless the implementation does so. That is why I urged to make this more explicit. FMPOV the first sentence should be removed as it suggests that something is happening *automagically* here ...

Reply:
Well, "Sends a command to the downstream peer" is also not happening unless the implementation does so.

From the viewpoint of someone *invoking* this interface method, I'm interested in what the method does - send a command *and* apply the result to the command context. That is an important piece of information for *invoking* the method, to emphasize that the command context doesn't need to get "accepted" by the invoking method afterwards.

Therefore I think this kind of target audience of the javadoc should be addressed here, not only the other viewpoint of the interface implementer.

(I initially wanted to not have the "apply the result to the command context" logic in that method, but the alternative would be passing the resulting proton delivery in some result value object, making things more complicated for the proton-independent interface.)

Reply:
I see your point.